src/core/backend_context.h

```diff
@@ -138,11 +138,12 @@ class BackendResponder {
       const std::vector<std::unique_ptr<InferenceRequest>>& requests,
       std::vector<std::unique_ptr<InferenceResponse>>* responses,
       const int max_batch_size, const bool pinned_enabled, cudaStream_t stream,
-      cudaEvent_t event)
+      cudaEvent_t event, bool zero_copy_support = false)
```

Comment:
I understand how this completely simplifies the logic but it exposes the zero_copy_support to the backend utilities API. Hence, this will be further exposed to other backends. We must be careful with naming. zero_copy_support in generic context shows there are no copies involved.. However, there is still a "memcpy" being enqueued to the stream.

I am okay for now but we must reconsider the nomenclature here. Probably as part of my changes to update backend utilities again with the core backend_context.h.

Reply:
Yes I agree. While the generic feature is zero copy support, we are not leveraging this completely since we still need the copies to prepare the contiguous buffers. 

I will consider renaming it to integrated_map_host_memory_support_ to be more accurate?

Reply:
From the responder's perspective, this variable is to determine whether a system `memcpy` is queued in the given copy stream, so maybe something like `copy_on_stream`

Reply:
@GuanLuo I am open to that name as well. It is more accurate to the behavior

Reply:
^^^ change the name here and elsewhere